### PR TITLE
Object Storage changes

### PR DESCRIPTION
**Pull request by maguiarr**
Using following environment variables:

- OBJECT_STORAGE_URL
- OBJECT_STORAGE_ACCESS_ID
- OBJECT_STORAGE_SECRET
- INSTANCE_URL_PREFIX

Files can be stored and retrieved from Object Storage
Deleting file from Object Storage if there's one there already of the PUBLIC_NOTICE type. Just following what's been done for database logic

### CODE REVIEW

File: apps/api/src/app/modules/attachment/attachment.service.ts [hunk 1]
```diff
@@ -47,14 +50,36 @@ export class AttachmentService extends DataService<Attachment, Repository<Attach
...
 
         // Now that the public notice is deleted, we can proceed with the regular creation.
       }
     }
 
-    return super.create(request, user);
+    // Starting changes for Object Store
+    const contentFile: Buffer = request.fileContents;
+
+    request.fileContents = Buffer.from('hello world', 'utf8');
```
Comment (basilv) [inline apps/api/src/app/modules/attachment/attachment.service.ts:63]: This should be flagged with a todo comment to remove this once we remove fileContents. In fact, given that you are already not using the column, you could configure the entity to not map the column, and clean this code up.

File: apps/api/src/app/modules/attachment/attachment.service.ts [hunk 2]
```diff
@@ -134,11 +159,64 @@ export class AttachmentService extends DataService<Attachment, Repository<Attach
...
+    const finalBuffer = await this.stream2buffer(dataStream);
+
+    attachmentFileResponse.fileContents = finalBuffer;
+
     return attachmentFileResponse;
+
   }
 
+  createObjectUrl(projectId: number, attachmentId: number, fileName: string): string {
+    if(process.env.INSTANCE_URL_PREFIX) {
```
Comment (basilv) [inline apps/api/src/app/modules/attachment/attachment.service.ts:179]: Should check if INSTANCE_URL_PREFIX is length 0 as well (undefined or length 0 means don't use it)

File: apps/api/src/minio.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+// Create the environment variables locally
+export const minioClient =  new Minio.Client({
+    endPoint:  process.env.OBJECT_STORAGE_URL,
+    accessKey: process.env.OBJECT_STORAGE_ACCESS_ID,
+    secretKey: process.env.OBJECT_STORAGE_SECRET
+});
+
+export const bucket = process.env.OBJECT_STORAGE_BUCKET;
+
+minioClient.listBuckets(function(err, buckets) {
```
Comment (basilv) [inline apps/api/src/minio.ts:12]: Why are we doing this? Looks like it could be removed.

File: package.json [hunk 4]
```diff
@@ -60,6 +60,10 @@
     "@angular/platform-browser": "^11.2.14",
     "@angular/platform-browser-dynamic": "^11.2.14",
     "@angular/router": "^11.2.14",
+    "@auth0/angular-jwt": "^5.0.2",
```
Comment (basilv) [inline package.json:63]: There's a lot of extra libraries as new dependencies - this doesn't look right. Should just be adding minio I believe.

**Review by basilv — CHANGES_REQUESTED** [review]

File: package.json (hunk 4 above)
Comment (maguiarr) [reply package.json:63]: All I did was to branch from release/FOM-12 and then run npm run install. I actually cloned the entire nr-fom-api. I found strange too that so many libraries were installed. Maybe you'll have the same result if you run npm run install.

**Pull request merged**